lib/UbersmithClient.js

```diff
@@ -56,10 +63,37 @@ class UbersmithClient {
                 if (error) {
                     reject(error);
                 } else {
-                    if (body.status) {
-                        resolve(body.data);
+                    if (response.statusCode == 200) { // Valid Ubersmith response
+                        // Responses are returned in JSON format (with the exception of specialized methods which return raw PDF, Image, XML or HTML data), and include the standard elements:
+                        //     status - API call success true/false
+                        //     error_code - error code for failed API calls
+                        //     error_message - error message for failed API calls
+                        //     data - results of API call
+                        //
+                        // Example Output
+                        // {
+                        //     "status":true,
+                        //     "error_code":null,
+                        //     "error_message":"",
+                        //     "data": "<api call output here>"
+                        // }
+                        if (body.status) {
+                            resolve(body.data);
+                        } else {
+                            reject(new UbersmithAPICallFailure(body));
+                        }
+                    } else if (response.statusCode < 200) {
```

Comment:
Handling this is part of the API response handling. I'm not sure I would use the suggested approach in a different language.

Maybe have a method here for handling this (not unlike the python-ubersmithclient)?